fmn/cache/rules.py

```diff
@@ -0,0 +1,52 @@
+import logging
+from typing import TYPE_CHECKING
+
+from cashews import cache
+from cashews.formatter import get_templates_for_func
+
+from ..database.model import Rule
+from .util import cache_early_ttl, cache_ttl
+
+if TYPE_CHECKING:
+    from fedora_messaging.message import Message
+    from sqlalchemy.ext.asyncio import AsyncSession
+
+log = logging.getLogger(__name__)
+
+
+class RulesCache:
+    """Cache the rules currently in the database."""
+
+    def __init__(self):
+        self._db: "AsyncSession" | None = None
+
+    def set_db(self, db: "AsyncSession"):
+        self._db = db
```

Comment:
Why not simply manipulate/use the member directly here (and call it `self.db`)?

Reply:
Yeah, I wasn't sure. I suppose I could do that as well. It feels more explicit and less hackish to go through a function call instead of manipulating the attribute directly, but maybe that's just me.
I'll change it to `.db`

Reply:
I imagine it’s idiomatic in Javascript e.a. to use accessor methods across the board? Python looks like an outlier here, with the convention to access members directly and if you need to intercept setting/getting, then turn it into a property.